e2e/test/attestations_test.go

```diff
@@ -102,11 +103,17 @@ func TestApprovedValUpdates(t *testing.T) {
 	})
 }
 
-func fetchAllAtts(ctx context.Context, cprov cchain.Provider, chainVer xchain.ChainVersion) ([]xchain.Attestation, error) {
+func fetchAllAtts(ctx context.Context, cprov cchain.Provider, chainVer xchain.ChainVersion, nodeIsDelayed bool) ([]xchain.Attestation, error) {
 	fromOffset := uint64(1) // Start at initialXBlockOffset
 	var resp []xchain.Attestation
 	for {
 		atts, err := cprov.AttestationsFrom(ctx, chainVer, fromOffset)
+		if err != nil && strings.Contains(err.Error(), "no commit info found") && nodeIsDelayed {
```

Comment:
feels like we can improve the interface or error which indicates "attestation pruned from historical state". Having to check for "no commit info found" isn't obvious.

What about some helper method like:
```
if cchain.IsErrHistoryPruned(err) {
  fromOffset++
}
```